docs/en/engines/table-engines/integrations/postgresql.md

```diff
@@ -0,0 +1,102 @@
+---
+toc_priority: 8
+toc_title: PostgreSQL
+---
+
+# PosgtreSQL {#postgresql}
+
+The PostgreSQL engine allows you to perform `SELECT` queries on data that is stored on a remote PostgreSQL server.
+
+## Creating a Table {#creating-a-table}
+
+``` sql
+CREATE TABLE [IF NOT EXISTS] [db.]table_name [ON CLUSTER cluster]
+(
+    name1 [type1] [DEFAULT|MATERIALIZED|ALIAS expr1] [TTL expr1],
+    name2 [type2] [DEFAULT|MATERIALIZED|ALIAS expr2] [TTL expr2],
+    ...
+) ENGINE = PostgreSQL('host:port', 'database', 'table', 'user', 'password');
+```
+
+See a detailed description of the [CREATE TABLE](../../../sql-reference/statements/create/table.md#create-table-query) query.
+
+The table structure can differ from the original PostgreSQL table structure:
+
+-   Column names should be the same as in the original PostgreSQL table, but you can use just some of these columns and in any order.
+-   Column types may differ from those in the original PostgreSQL table. ClickHouse tries to [cast](../../../sql-reference/functions/type-conversion-functions.md#type_conversion_function-cast) values to the ClickHouse data types.
+
+**Engine Parameters**
+
+-   `host:port` — PostgreSQL server address.
+
+-   `database` — Remote database name.
+
+-   `table` — Remote table name.
+
+-   `user` — PostgreSQL user.
+
+-   `password` — User password.
+
+SELECT Queries on PostgreSQL side run as `COPY (SELECT ...) TO STDOUT` inside read-only PostgreSQL transaction with commit after each `SELECT` query.
+
+Simple `WHERE` clauses such as `=, !=, >, >=, <, <=, IN` are executed on the PostgreSQL server.
+
+All joins, aggregations, sorting, `IN [ array ]` conditions and the `LIMIT` sampling constraint are executed in ClickHouse only after the query to PostgreSQL finishes.
+
+INSERT Queries on PostgreSQL side run as `COPY "table_name" (field1, field2, ... fieldN) FROM STDIN` inside PostgreSQL transaction with auto-commit after each `INSERT` statement.
+
+PostgreSQL Array types converts into ClickHouse arrays.
```

Comment:
Also in postgres an array data type column like Integer[] may contain arrays of different dimensions in different rows, but in ClickHouse only allowed multidimensional arrays of the same dimension in all rows.

Reply:
fixed in https://github.com/ClickHouse/ClickHouse/pull/21078/commits/aacc9c091f983454faa04526b13b9792006684b6